shared/util/make-reducer.tsx

```diff
@@ -12,6 +12,7 @@ export type ActionHandler<A, S> = {
 
 function makeReducer<A, S>(initialState: S, map: ActionHandler<A, S>) {
   return (state: S = initialState, action: TypedActions): S => {
+    // @ts-ignore
```

Comment:
what's broken here?

Reply:
It can't correctly refine here (nor has it ever). We explicitly say what the input and output types are so it's fine